dashboard/src/actions/overviewActions.js

```diff
@@ -115,22 +121,31 @@ export const updateDataset =
       dispatch({ type: TYPES.LOADING });
 
       const runs = getState().overview.datasets;
-
-      const method = metaDataActions[actionType];
-
       const endpoints = getState().apiEndpoint.endpoints;
+      let method;
+      const isMultipleUpdate = Array.isArray(actionType);
+      if (!isMultipleUpdate) {
+        method = metaDataActions[actionType];
+      }
+
+      const payload = isMultipleUpdate
+        ? actionType.reduce((r, c) => Object.assign(r, c), {})
+        : { [method]: actionValue };
```

Comment:
This is annoyingly complex (and potentially confusing).

My understanding of your proposed change is that it allows `updateDataset()` to accept either an action type/value pair or an array of pairs; it then builds an object either from the pair or from the reduction of all the pairs; and then it does a `PUT` of that object.

I presume that you want the function here to continue to accept a separate pair so that you don't have to modify all the callers of the code...which is fine.  But, suppose that, instead of introducing this complexity here, we refactor the original function and add two functions on top of it:
- the interface of the original function is unchanged (and it continues to accept only a single pair), but its body is changed to create the object from the pair and pass it to a new subroutine;
- the new subroutine contains the remainer of the body of the original function; and
- add a new function which accepts an array of pairs, which reduces them into a single object and passes it to the new subroutine.

In this way, no existing code needs to be changed, and we don't need conditionals to figure out what the arguments were.

And, potentially, you won't actually need the third function -- the caller which intends to pass the array in for reduction could instead simply construct an object instead of an array and pass it directly to the new subroutine.

Reply:
Updated as suggested